### PR TITLE
Revert profiler behavior. Do not record trivial functions that are ca…

### PR DESCRIPTION
**Pull request by mwootton**
…lled often.

When the profiler was moved to a dispatcher it stopped honoring a set of DONT_PROFILE functions.  This inflates the number for recorded functions calls (by a factor of about 2x in my tests).  This balloons of the size of the json and table output making it harder to load, e.g in chrome tracing.

This change filters to the original list of functions.

### CODE REVIEW

**Comment by colesbury** [comment]
@albanD can you review this?

@mwootton is there an issue to track this regression?

**Comment by mwootton** [comment]
This pr isn't in response to a specific issue.  One could have been made independently of course.

**Comment by mwootton** [comment]
Also, the regression happened between these commits:
1592d6842cc8d4fabca35a6f9ac3117c34348e70 [resubmit] Move profiler to a dispatch wrapper (#36766)                                 bcdb0727c263adf11d1f26a5457ba756b76baa7f Revert D20907254: Fix long line splitting issue in python_print

**Comment by sunway513** [comment]
@mwootton  can you rebase your PR to resolve the conflicts and check the CI failures?
cc @jeffdaily

**Comment by jeffdaily** [comment]
@albanD (CC @colesbury) gentle ping.  Can we get a review?

File: tools/autograd/gen_variable_type.py [hunk 1]
```diff
@@ -717,6 +723,7 @@ def emit_profiled_body(declaration):
     modifies_arguments = inplace or is_out_fn
     returns_void = len(returns) == 0
 
+    base_name = name[:-1] if inplace else name[:-4] if is_out_fn else name
```
Comment (albanD) [inline tools/autograd/gen_variable_type.py:726]: You should already have access to the base name of the function without doing this in this file. Something like `declaration["base_name"]` ?

File: test/test_autograd.py [hunk 2]
```diff
@@ -2666,9 +2666,9 @@ def test_profiler(self):
         self.assertFalse(torch.autograd._profiler_enabled())
 
         last_end = 0
-        names = ['is_complex', 'mul', 'to', 'empty_strided', 'copy_', 'is_complex',
+        names = ['is_complex', 'mul', 'to', 'empty_strided', 'is_complex',
```
Comment (albanD) [inline test/test_autograd.py:2669]: Why are the copies gone from here? We actually would like to keep these.
Do we need to update the TracingTypeManual implementation to make sure copy_ is still properly tracked?

**Pull request closed without merging**